Mathlib/LinearAlgebra/TensorProduct/Decomposition.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2025 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import Mathlib.Algebra.DirectSum.Decomposition
+import Mathlib.LinearAlgebra.DirectSum.TensorProduct
+
+/-! # Decomposition of tensor product
+
+In this file we show that if `ℳ` is a graded `R`-module, and `S` is any `R`-algebra, then
+`S ⊗[R] ℳ` (which is actually `fun i ↦ (ℳ i).baseChange S`) is a graded `S`-module with the same
+grading.
```

Comment:
This file is about decomposition and not gradings (i.e. decomposition indexed by a monoid that commutes with the grading on `R`)